### PR TITLE
Improve css error reporting

### LINKED ISSUE #3569: CSS errors have a broken path
CSS errors will create an invalid path to the CSS, if the CSS is in code. This is because it adds the widget name to the path.

Here's an example:

Image

If you click the heading it should take you to the file containing the error, but it doesn't because the path is invalid.

It would be nice to display the widget within the title, but it should probably be a separate field.

### PR DESCRIPTION
**Pull request by rodrigogiraoserrao**
This PR fixes #3569.

This also improves error importing by providing the widget name/class variable from where CSS was read in case the CSS is inline CSS.
This is still a draft because I still need to figure out if we can link to a specific line/column for when the CSS comes from an external file.

### CODE REVIEW

File: src/textual/css/_styles_builder.py [hunk 1]
```diff
@@ -65,19 +65,6 @@
...
-    Args:
-        tokens: Tokens to join
-        joiner: String to join on.
-
-    Returns:
-        The tokens, joined together to form a string.
-    """
-    return joiner.join(token.value for token in tokens)
-
-
```
Comment (rodrigogiraoserrao) [inline src/textual/css/_styles_builder.py:80]: This was dead code.

File: src/textual/dom.py [hunk 2]
```diff
@@ -416,33 +416,41 @@ def __rich_repr__(self) -> rich.repr.Result:
...
             try:
-                return f"{getfile(base)}:{base.__name__}"
+                return (getfile(base), f"{base.__name__}.DEFAULT_CSS")
             except (TypeError, OSError):
-                return f"{base.__name__}"
+                return ("", f"{base.__name__}.DEFAULT_CSS")
 
         for tie_breaker, base in enumerate(self._node_bases):
-            css: str = base.__dict__.get("DEFAULT_CSS", "").strip()
+            css: str = base.__dict__.get("DEFAULT_CSS", "")
```
Comment (rodrigogiraoserrao) [inline src/textual/dom.py:448]: This messed up the line/column reporting of inline CSS errors.

File: src/textual/app.py [hunk 3]
```diff
@@ -1753,20 +1753,19 @@ def _load_screen_css(self, screen: Screen):
 
         update = False
         for path in screen.css_path:
-            if not self.stylesheet.has_source(path):
+            if not self.stylesheet.has_source((str(path), "")):
```
Comment (willmcgugan) [inline src/textual/app.py:1756]: I'd like to veto the tuple here. It doesn't read well. Could the second value be a keyword arg?

File: src/textual/css/stylesheet.py [hunk 4]
```diff
@@ -68,27 +68,35 @@ def __rich_console__(
         for token, message in errors:
             error_count += 1
 
-            if token.path:
-                path = Path(token.path)
-                filename = path.name
+            if token.read_from:
```
Comment (willmcgugan) [inline src/textual/css/stylesheet.py:71]: Could this be an explicit `is not None` check.

**Review by willmcgugan — CHANGES_REQUESTED** [review]
Looks good. Couple of requests.

Can you add a screenshot of how these changes look in CSS errors?

File: src/textual/app.py (hunk 3 above)
Comment (rodrigogiraoserrao) [reply src/textual/app.py]: Yeah, that sounds good. See 6c6eecf.

File: src/textual/css/stylesheet.py (hunk 4 above)
Comment (rodrigogiraoserrao) [reply src/textual/css/stylesheet.py]: Reworked this a bit better so that the checks make more sense.

**Comment by davep** [comment]
Giving this spin with this code:

```python
from textual.app import App, ComposeResult
from textual.widgets import Label

class CSSErrorApp(App[None]):

    CSS = """
    :
    """

    def compose(self) -> ComposeResult:
        yield Label()

if __name__ == "__main__":
    CSSErrorApp().run()
```

I get this error:

```
 Error in stylesheet:
 /Users/davep/develop/python/textual-upstream/sandbox/foo.py, CSSErrorApp.CSS:1:4
╭────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│   1 │                                                                                                                                                                                                                      │
│ ❱ 2 │   :                                                                                                                                                                                                                  │
│   3                                                                                                                                                                                                                        │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
 • Expected one of 'comment line', 'comment start', 'selector start', 'selector start class', 'selector start id', 'selector start universal', 'variable name', or 'whitespace'.                                              
 • Did you forget a semicolon at the end of a line?        
```

This *seems* to be saying that the error is on line 1, but also on line 2. Am I reading that correctly?

**Comment by rodrigogiraoserrao** [comment]
> This _seems_ to be saying that the error is on line 1, but also on line 2. Am I reading that correctly?

You're reading that correctly, yes, but that also seems to be an issue in `master`.
Should I open an issue to log this so that this PR is independent of that bug?

**Comment by davep** [comment]
Cool, if it's a pre-existing thing unrelated this makes sense it's handled in a different fix.

**Review by davep — APPROVED** [review]

**Pull request merged**